platooning_strategic_IHP/src/platoon_strategic_ihp.cpp

```diff
@@ -283,830 +168,541 @@ namespace platoon_strategic
+         *       The "platooning info status" reflect the overall operating status. 
+         *       The "platoon strategic plugin states" manage the negotiation strategies and vehicle communication in a more refined manner. 
+         * A more detailed note can be found in the corresponding function declaration in "platoon_strategic_ihp.h" file.
+         */
+
+        cav_msgs::PlatooningInfo status_msg;
+
+        if (pm_.current_platoon_state == PlatoonState::STANDBY)
         {
-            mobility_response = mob_req_cb_leader(msg);
+            status_msg.state = cav_msgs::PlatooningInfo::DISABLED;
         }
-        else if (pm_.current_platoon_state == PlatoonState::FOLLOWER)
+        else if (pm_.current_platoon_state == PlatoonState::LEADER)
         {
-            mobility_response = mob_req_cb_follower(msg);
+            status_msg.state = pm_.getTotalPlatooningSize() == 1 ? cav_msgs::PlatooningInfo::SEARCHING : cav_msgs::PlatooningInfo::LEADING;
+        }
+        else if (pm_.current_platoon_state == PlatoonState::LEADERWAITING)
+        {
+            status_msg.state = cav_msgs::PlatooningInfo::CONNECTING_TO_NEW_FOLLOWER;
         }
         else if (pm_.current_platoon_state == PlatoonState::CANDIDATEFOLLOWER)
         {
-            mobility_response = mob_req_cb_candidatefollower(msg);
+            status_msg.state = cav_msgs::PlatooningInfo::CONNECTING_TO_NEW_LEADER;
         }
-        else if (pm_.current_platoon_state == PlatoonState::LEADERWAITING)
+        else if (pm_.current_platoon_state == PlatoonState::FOLLOWER)
         {
-            mobility_response = mob_req_cb_leaderwaiting(msg);
+            status_msg.state = cav_msgs::PlatooningInfo::FOLLOWING;
         }
-        else if (pm_.current_platoon_state == PlatoonState::STANDBY)
+        // UCLA: add leader aborting for frontal join (inherited from candidate follower)
+        else if (pm_.current_platoon_state == PlatoonState::LEADERABORTING)
         {
-            mobility_response = mob_req_cb_standby(msg);
+            status_msg.state = cav_msgs::PlatooningInfo::CONNECTING_TO_NEW_LEADER;
         }
+        // UCLA: add candidate leader for frontal join (inherited from leader waiting)
+        else if (pm_.current_platoon_state == PlatoonState::CANDIDATELEADER)
+        {
+            status_msg.state = cav_msgs::PlatooningInfo::CONNECTING_TO_NEW_FOLLOWER;
+        }
+        
+        // compose msgs for standby
+        if (pm_.current_platoon_state != PlatoonState::STANDBY)
+        {
+            status_msg.platoon_id = pm_.currentPlatoonID;
+            status_msg.size = std::max(1, pm_.getTotalPlatooningSize());
+            status_msg.size_limit = config_.maxPlatoonSize;
 
-        return mobility_response;
+            if (pm_.current_platoon_state == PlatoonState::FOLLOWER)
+            {
+                ROS_DEBUG_STREAM("isFollower: " << pm_.isFollower);
+                ROS_DEBUG_STREAM("pm platoonsize: " << pm_.platoon.size());
```

Comment:
Why does this line use pm_.platoon.size() and 5 lines above uses pm_.getTotalPlatooningSize() for presumably the same info?  If they are different it would be good to include a comment explaining why we need different approaches.  If they are the same, I suggest using the former (accessor call) to avoid directly reading what should be a private member object.